src/definitions/x509/crl.rs

```diff
@@ -0,0 +1,319 @@
+                        .map_err(|_| Error::InvalidPublicKeyFormat)?;
+
+                    let signature = p256::ecdsa::Signature::from_slice(signature)
+                        .map_err(|_| Error::SignatureInWrongFormat)?;
+
+                    key.verify(&crl_raw, &signature)
+                        .map_err(|_| Error::SignatureCheckOfCrlFailed)?;
+
+                    Ok(crl.tbs_cert_list)
+                }
+                CurveKind::P384 => {
+                    let key = p384::ecdsa::VerifyingKey::from_sec1_bytes(&sec1_public_key)
+                        .map_err(|_| Error::InvalidPublicKeyFormat)?;
+
+                    let signature = p384::ecdsa::Signature::from_slice(signature)
+                        .map_err(|_| Error::SignatureInWrongFormat)?;
+
+                    key.verify(&crl_raw, &signature)
+                        .map_err(|_| Error::SignatureCheckOfCrlFailed)?;
+
+                    Ok(crl.tbs_cert_list)
+                }
+            }
+        }
+        _ => Err(Error::UnknownSignatureAlgorithm(Box::new(
+            crl.signature_algorithm,
+        ))),
+    }
+}
+
+fn decode_cert_list(bytes: &[u8]) -> Result<(Vec<u8>, CertificateList), Error> {
+    let (_, top_sequence) =
+        Sequence::from_der(bytes).map_err(Error::ExtractCrlSequenceSignature)?;
+
+    let top_sequence_content = top_sequence.into_content();
+    let top_sequence_bytes = top_sequence_content.as_ref();
+
+    let (_, cert_list_sequence) =
+        Sequence::from_der(top_sequence_bytes).map_err(Error::ExtractCrlSequenceSignature)?;
+
+    let cert_list_content = cert_list_sequence.into_content();
+    let cert_list_bytes = cert_list_content.as_ref();
+
+    let cert_list = CertificateList::from_der(bytes)
+        .map_err(|err| Error::ParsingCrlComponent("cert list", err))?;
+
+    Ok((cert_list_bytes.to_vec(), cert_list))
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum CurveKind {
+    P256,
+    P384,
+}
+
+impl TryFrom<Option<&Any>> for CurveKind {
+    type Error = Error;
+
+    fn try_from(ext_bit_string: Option<&Any>) -> Result<Self, Self::Error> {
+        let ext_any = ext_bit_string.ok_or(Error::MissingCurve)?;
+        let obj_id = ObjectIdentifier::from_der(ext_any.value())
+            .map_err(|err| Error::ParsingCrlComponent("curve kind extension", err))?;
+
+        let curve = match obj_id {
+            OID_EC_CURVE_P256 => Self::P256,
+            OID_EC_CURVE_P384 => Self::P384,
+            other => return Err(Error::UnknownCurveExt(other)),
+        };
+
+        Ok(curve)
+    }
+}
+
+/// Given revocation `cert_lists`, check if `cert` has been revoked
+pub fn check_cert_against_cert_lists(
+    cert: &TbsCertificate,
+    cert_lists: &[TbsCertList],
+) -> Result<(), Error> {
+    for cert_list in cert_lists {
+        if cert.issuer != cert_list.issuer {
+            return Err(Error::IssuerMismatchBetweenCertAndCrl);
```

Comment:
Is there a requirement for these to match?

Reply:
RFC 5280 6.3.1. CRL Processing (b)(1)

> [V]erify that the CRL issuer matches the certificate issuer.